packages/interactive-messages/src/adapter.ts

```diff
@@ -573,22 +676,35 @@ export interface OptionsConstraints {
   within: 'block_actions' | 'interactive_message' | 'dialog';
 }
 
-type AnyConstraints = ActionConstraints | OptionsConstraints;
+/**
+ * Constraints on when to call a view submission or view closed handler.
+ */
+export interface ViewConstraints {
+  callbackId?: string | RegExp;
+  externalId?: string;
```

Comment:
Should we allow `RegExp` for `externalId` too? Since they are unique on a per-user basis, it would be really hard to setup a handler for them with a type of string.

Reply:
Earlier I was trying not to expand scope so I didn't add RegExp matching to anything new - but now that you point it out that totally makes sense as a use case. It's not too much work to add that in now, so I might as well.